Only members

GzimAsani opened the pull request:
Members-only
In this project we:

- Use devise to provide users authentication
- Use devise to create custom view
- New member can create secret posts
- Hide post authors when not logged in
- Members can only edit or delete their own posts.

GzimAsani committed:
First commit

GzimAsani committed:
Application gems dependencies

GzimAsani committed:
Add migration files

GzimAsani committed:
Created functionality, CRUD operations

GzimAsani committed:
The following changes were made:
I created edit, new, delete functionalities
I added bootstrap
I added links into my rootpage

GzimAsani committed:
Clean up layout

GzimAsani committed:
devise user model

GzimAsani committed:
Add and improved Devise paths
Forgot password
Log in
Sign up

GzimAsani committed:
Add those restrictions:
Only the users can see the article creators
Only users can create articles
Current users can delete/edit only their respective articles
Improve layout pages

GzimAsani committed:
Updated Readme

GzimAsani committed:
Updated readme, add pictures

GzimAsani committed:
Fix rubocop issues with rubocop-a

GzimAsani committed:
Fix linter errors

GzimAsani committed:
Fix linters

GzimAsani committed:
Fix linter errors

lekegitrepo commented on config/routes.rb:
```diff
@@ -0,0 +1,5 @@
+Rails.application.routes.draw do
+  devise_for :users, :controllers => { registrations: 'registrations' }
+  root 'pages#home'
+  resources :blogs
```
- **_The application routes should be `resources :blogs, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). Mainly you should specify only the useful routes._**

lekegitrepo commented on app/controllers/blogs_controller.rb:
```diff
@@ -0,0 +1,74 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
```
- **_There should be a `before_action` that restrict access to the `:new` and `:create` methods ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). This functionality might have been covered by `before_action :authenticate_user!` however the requirement specifically mentioned `before_action` filter for `new` and `create` methods_**

lekegitrepo commented on app/controllers/blogs_controller.rb:
```diff
@@ -0,0 +1,74 @@
+  def show; end
+
+  # GET /blogs/new
+  def new
+    @blog = current_user.blogs.build
+  end
```
- **_It should initialize an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

lekegitrepo reviewed (changes requested):
## Status: Changes Required :recycle:
### Hi @GzimAsani :clap: :clap:
#### This is a good project :+1: and you really did a good on the application view/user interface. However, some requirements are not implemented as specified in the Project Specification list. Check my comments below :point_down:

#### Login Page:
- **_Change the login button from `Sign up` to `Sign in`_**

#### PostsController(or in this case BlogsController):
- **_Implement the methods are mentioned in the specification_**

#### routes:
- **_Specified the useful routes_**


#### _Reviewed by Temitayo Adeleke_

GzimAsani committed:
I made the following changes based on the reviews
I changed revised the controller
Deleted the edit rout
Deleted the destroy rout
Deleted the show action

tirthajyoti-ghosh commented on app/controllers/blogs_controller.rb:
```diff
@@ -0,0 +1,74 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```
You need to create a custom `before_action` method to restrict users to specific actions and that custom method should be declared in `ApplicationController` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)).

```ruby
before_action :signed_in_only!, only: [:new, :create]

def signed_in_only!
  # current_user in ApplicationController must be defined as helper method to make it work
  redirect_to  new_user_session_path unless current_user
end
```

tirthajyoti-ghosh reviewed (changes requested):
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- Please remove the `node_modules`, `tmp`, and `log` folders from your repository and add it to `.gitignore`.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

GzimAsani committed:
following changes were made based on reviews:
Removed log, tmp and node_mudoles folders and add them to my gitignore
Add a redirection path in ApplicationController

GzimAsani committed:
Changes from rubocop-a

lekegitrepo reviewed (changes requested):
## Status: Changes Required :recycle:
#### Good job so far :+1: However, as much as I wanted to approve this project you are missing this `before_action :signed_in_only!, only: [:new, :create]`. Implement this and submit for another review.

#### _Reviewed by Temitayo Adeleke_

GzimAsani committed:
Add changes based on the last review

GzimAsani committed:
Fix rubocop -a

Dandush03 reviewed (approved):
# Status: :rocket: Approved 
## Great Work! keep it up
### Since none of the preview reviews ask you to, I'll leave it here as optional but it's a must. so please add it to your project before you  move on,
- [ ] Add .gitignore file and remove all files from your git-flow; I'll leave an example below:
```bash
# See https://help.github.com/articles/ignoring-files for more about ignoring files.
#
# If you find yourself ignoring temporary files generated by your text editor
# or operating system, you probably want to add a global ignore instead:
#   git config --global core.exclude file '~/.gitignore_global'

# Ignore bundler config.
/.bundle

# Ignore all logfiles and tempfiles.
/log/*
/tmp/*
!/log/.keep
!/tmp/.keep

# Ignore schema
/db/schema.rb

# Ignore pidfiles, but keep the directory.
/tmp/pids/*
!/tmp/pids/
!/tmp/pids/.keep

# Ignore uploaded files in development.
/storage/*
!/storage/.keep

/public/assets
.byebug_history

# Ignore master key for decrypting credentials and more.
/config/credentials/*.key

/public/packs
/public/packs-test
/node_modules
/yarn-error.log
yarn-debug.log*
.yarn-integrity

/config/credentials/development.key

```

# Happy Coding :tophat: :nerd_face:
